### PR TITLE
Improve APIs of image io to support passing html id and element

### LINKED ISSUE #160: Improve cv.imread and cv.VideoCapture
Now cv.imread only supports canvas id and cv.VideoCapture only supports video id.
If cv.imread could support canvas element, img id and img element, and cv.VideoCapture could support video element, it will be more user-friendly.

### PR DESCRIPTION
**Pull request by pancx**
Support passing canvas id or element and img id or element to cv.imread.
Support passing canvas element or id to cv.imshow.
Support passing video element or id to cv.VideoCapture.
fix #160

### CODE REVIEW

**Comment by pancx** [comment]
@huningxin  Please have a look.

File: modules/js/src/helpers.js [hunk 1]
```diff
@@ -1,18 +1,39 @@
 Module["imread"] = function(canvasID) {
-    var canvas = document.getElementById(canvasID);
-    if (canvas === null || !(canvas instanceof HTMLCanvasElement)) {
-        throw("Please input the valid canvas id.");
+    var img = null;
```
Comment (huningxin) [inline modules/js/src/helpers.js]: "canvasID" name now seems a bit confusing. Please use some general names.

File: modules/js/src/helpers.js [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
+    var canvas = null;
+    var ctx = null;
+    if (img instanceof HTMLImageElement) {
+        canvas = document.createElement("canvas");
+        canvas.width = img.width;
+        canvas.height = img.height;
+        ctx = canvas.getContext("2d");
+        ctx.drawImage(img, 0, 0, img.width, img.height);
+    }
+    else {
```
Comment (huningxin) [inline modules/js/src/helpers.js]: put the else to one line above, like
```
} else {
```

File: modules/js/src/helpers.js [hunk 3]
```diff
@@ -1,18 +1,39 @@
...
+        canvas = img;
+    }
+    if (!(canvas instanceof HTMLCanvasElement)) {
+        throw("Please input the valid canvas or img id.");
         return;
     }
-    var ctx = canvas.getContext("2d");
+    
+    ctx = canvas.getContext("2d");
```
Comment (huningxin) [inline modules/js/src/helpers.js]: If input is an image, `ctx` is already gotten.

**Review by huningxin — CHANGES_REQUESTED** [review]

**Comment by pancx** [comment]
@huningxin Updated. Please have a look.

**Review by huningxin — APPROVED** [review]

**Comment by huningxin** [comment]
Looks good. Thanks!

**Pull request merged**